### PR TITLE
[recusal-list] Add strategy optional param

### PR DESCRIPTION
**Pull request by 0x0-github**
Fixes # .

Changes proposed in this pull request:
- Addition of erc721-excluding-addresses strategy
- 
-

### CODE REVIEW

File: src/strategies/erc721-excluding-addresses/index.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    network,
+    provider,
+    abi,
+    addresses.filter((address: any) => !options.excluded.includes(address))
+      .map((address: any) => [options.address, 'balanceOf', [address]]),
+    { blockTag }
+  );
+  return Object.fromEntries(
+    response.map((value, i) => [
+      addresses[i],
```
Comment (ChaituVR) [inline src/strategies/erc721-excluding-addresses/index.ts:30]: addresses will have different index than response, as you are filtering the addresses above

File: src/strategies/erc721-excluding-addresses/index.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import { formatUnits } from '@ethersproject/units';
+import { multicall } from '../../utils';
+
+export const author = '0x0';
```
Comment (ChaituVR) [inline src/strategies/erc721-excluding-addresses/index.ts:4]:
```suggestion
export const author = '0x0-github';
```

**Review by ChaituVR — CHANGES_REQUESTED** [review]
Hi @0x0-github maybe we can modify https://snapshot.org/#/strategy/recusal-list this strategy to pass  a new param `strategy`  like this https://github.com/snapshot-labs/snapshot-strategies/blob/master/src/strategies/anti-whale/index.ts#L69 will be more reusable

**Comment by 0x0-github** [comment]
Oh thanks indeed this would be better for future usages of anyone, can I just update it like that then ?
No need to mention my name then ?

**Comment by 0x0-github** [comment]
Updated the strategy (added optional strategy param and associated parameters as well), putted it to v 0.2.0 and removed the symbol parameter / definition (Doesn't seem to be used at all). Let me know if there's any issue / especially on the json definition file since I'm not a pro yet, some field may have more suitable config

File: src/strategies/recusal-list/README.md [hunk 3]
```diff
@@ -1,14 +1,21 @@
...
     "0xa478c2975ab1ea89e8196811f51a7b7ade33eb11",
     "0xeF8305E140ac520225DAf050e2f71d5fBcC543e7"
-  ]
+  ],
+  "strategy": {
+    "name": "erc721",
+    "params": {
+      "address": "0x3B8CeB26f4FabACbD02b22caeceeb26D67E4013A",
+      "symbol": "MLZ"
+  }
```
Comment (ChaituVR) [inline src/strategies/recusal-list/README.md:19]:
```suggestion
  "strategy": {
    "name": "erc721",
    "params": {
      "address": "0x3B8CeB26f4FabACbD02b22caeceeb26D67E4013A",
      "symbol": "MLZ"
    }
  }
```

File: src/strategies/recusal-list/index.ts [hunk 4]
```diff
@@ -1,21 +1,42 @@
...
+      space,
+      network,
+      provider,
+      addresses,
+      options.strategy.params,
+      snapshot
+    );
+
+    return Object.fromEntries(
+      Object.entries(result).map(([address, _]) => [
```
Comment (ChaituVR) [inline src/strategies/recusal-list/index.ts:29]:
```suggestion
      Object.entries(result).map(([address]) => [
```

File: src/strategies/recusal-list/index.ts [hunk 5]
```diff
@@ -1,21 +1,42 @@
...
     return address.toLowerCase();
   });
 
-  return Object.fromEntries(
-    addresses.map((address) => [
-      address,
-      recusalList.includes(address.toLowerCase()) ? 0 : 1
-    ])
-  );
+  if (options.strategy.name) {
```
Comment (ChaituVR) [inline src/strategies/recusal-list/index.ts:18]:
```suggestion
  if (options.strategy?.name) {
```

File: src/strategies/recusal-list/index.ts [hunk 6]
```diff
@@ -1,21 +1,42 @@
...
+    const result = await strategies[options.strategy.name].strategy(
+      space,
+      network,
+      provider,
+      addresses,
```
Comment (ChaituVR) [inline src/strategies/recusal-list/index.ts:23]:
```suggestion
      recusalList,
```

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: src/strategies/recusal-list/index.ts [hunk 7]
```diff
@@ -1,21 +1,42 @@
...
-      address,
-      recusalList.includes(address.toLowerCase()) ? 0 : 1
-    ])
-  );
+  if (options.strategy?.name) {
+    const result = await strategies[options.strategy.name].strategy(
+      space,
+      network,
+      provider,
+      recusalList,
```
Comment (ChaituVR) [inline src/strategies/recusal-list/index.ts:23]:
```suggestion
      addresses.filter((address: any) => !recusalList.includes(address.toLowerCase())),
```

**Comment by 0x0-github** [comment]
Sorry I just saw and finished to read your reviews. Thanks 🙏.

**Comment by ChaituVR** [comment]
Deleting schema.json for now until we fix https://github.com/snapshot-labs/snapshot/issues/262

**Review by ChaituVR — APPROVED** [review]

**Pull request merged**